middleware/logging_test.go

```diff
@@ -57,6 +58,51 @@ func TestBadWriteLogging(t *testing.T) {
 	}
 }
 
+func TestDisabledSuccessfulRequestsLogging(t *testing.T) {
+	for _, tc := range []struct {
+		err         error
+		disableLog  bool
+		logContains string
+	}{
+		{
+			err:         nil,
+			disableLog:  false,
+			logContains: "GET http://example.com/foo (200)",
+		}, {
+			err:         nil,
+			disableLog:  true,
+			logContains: "",
+		},
+	} {
+		buf := bytes.NewBuffer(nil)
+		logrusLogger := logrus.New()
+		logrusLogger.Out = buf
+		logrusLogger.Level = logrus.DebugLevel
+
+		loggingMiddleware := Log{
+			Log:                      logging.Logrus(logrusLogger),
+			DisableRequestSuccessLog: tc.disableLog,
+		}
+
+		handler := func(w http.ResponseWriter, r *http.Request) {
+			io.WriteString(w, "<html><body>Hello World!</body></html>") //nolint:errcheck
+		}
+		loggingHandler := loggingMiddleware.Wrap(http.HandlerFunc(handler))
+
+		req := httptest.NewRequest("GET", "http://example.com/foo", nil)
+		recorder := httptest.NewRecorder()
+
+		w := errorWriter{
+			err: tc.err,
+			w:   recorder,
+		}
+		loggingHandler.ServeHTTP(w, req)
+		content := buf.String()
+
+		require.True(t, strings.Contains(content, tc.logContains))
```

Comment:
This is a bit weak for the second test case.
Wouldn't it be better to require that it does _not_ contain the request?

Reply:
Done! WDYT?